Source/core/CyclicBuffer.cpp

```diff
@@ -430,22 +438,27 @@ namespace Core {
         uint32_t tail = oldTail & _administration->_tailIndexMask;
         uint32_t free = Free(_administration->_head, tail);
 
-        while (free <= required) {
-            uint32_t remaining = required - free;
-            Cursor cursor(*this, oldTail, remaining);
-            uint32_t offset = GetOverwriteSize(cursor);
-            ASSERT((offset + free) >= required);
+        if (free <= required) {
```

Comment:
The cyclic buffer implementation is based on optimistic concurrency, assume it works unless at the end when we check it, the numbers do not add up, than we repeat.
The free size, might by the time we reach the check point already be occupied again and thus free space can not be assured, the while loop will make sure we will try again untill we have what we expected. If the cursor (line 445) does not ensure enough free space, than that is the actual error.
So here I am interested which test case led to these changes.